### PR TITLE
modify the conditions as PythonModuleVariable

### PR DESCRIPTION
**Pull request by jiayisunx**
## Motivation
The current code of `value in [torch.backends.cudnn, torch.ops]` requires `value` to have the implementation of `__eq__`. If the value is a custom object and does not implement `__eq__`, dynamo will throw error. For example, ConvolutionOpContext, the custom 'torch._C.ScriptClass' object registered in IPEX, dynamo will throw the following error:

**torch._dynamo.exc.InternalTorchDynamoError: '__eq__' is not implemented for __torch__.torch.classes.ipex_prepack.ConvolutionOpContext**

I think this is a common issue, To avoid this issue, the PR replaces the current code `value in [torch.backends.cudnn, torch.ops]`with `isinstance(value, (torch.backends.cudnn.CudnnModule, torch._ops._Ops)`

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116856

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/builder.py [hunk 1]
```diff
@@ -695,8 +695,14 @@ def build_key_value(k, v):
             istype(value, (types.ModuleType, replay_record.DummyModule))
             # type(torch.backends.cudnn) -> <class 'torch.backends.cudnn.CudnnModule'>
             # type(torch.ops) -> <class 'torch._ops._Ops'>
-            or value in [torch.backends.cudnn, torch.ops]
```
Comment (Skylion007) [inline torch/_dynamo/variables/builder.py:698]: If this isn't working, wouldn't:
```suggestion
            or any(value is obj for obj in (torch.backends.cudnn, torch.ops))
```
work?

**Review by jansel — CHANGES_REQUESTED** [review]
This seems like a semantic change.  `isinstance` is not the same as `==`.

I'd suggest replacing it with `is` instread as @Skylion007 suggests.

File: torch/_dynamo/variables/builder.py (hunk 1 above)
Comment (jiayisunx) [reply torch/_dynamo/variables/builder.py:698]: It works, thanks!

**Comment by jiayisunx** [comment]
> This seems like a semantic change. `isinstance` is not the same as `==`.
> 
> I'd suggest replacing it with `is` instread as @Skylion007 suggests.

I have made the changes, please review again, thanks!

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**